src/char/int_pet.c

```diff
@@ -139,36 +143,47 @@ static int inter_pet_fromsql(int pet_id, struct s_pet *p)
 
 	//`pet` (`pet_id`, `class`,`name`,`account_id`,`char_id`,`level`,`egg_id`,`equip`,`intimate`,`hungry`,`rename_flag`,`incubate`, `autofeed`)
 
-	if( SQL_ERROR == SQL->Query(inter->sql_handle, "SELECT `pet_id`, `class`,`name`,`account_id`,`char_id`,`level`,`egg_id`,`equip`,`intimate`,`hungry`,`rename_flag`,`incubate`,`autofeed` FROM `%s` WHERE `pet_id`='%d'", pet_db, pet_id) )
-	{
-		Sql_ShowDebug(inter->sql_handle);
+	const char *query = "SELECT "
+		"`class`, `name`, `account_id`, `char_id`, `level`, `egg_id`, `equip`, `intimate`, `hungry`, `rename_flag`, `incubate`, `autofeed` "
+		"FROM `%s` WHERE `pet_id`=?";
+
+	if (SQL_ERROR == SQL->StmtPrepare(stmt, query, pet_db) ||
+	    SQL_ERROR == SQL->StmtBindParam(stmt, 0, SQLDT_UINT32, &pet_id, sizeof(pet_id)) ||
+	    SQL_ERROR == SQL->StmtExecute(stmt) ||
+	    SQL_ERROR == SQL->StmtBindColumn(stmt, 0, SQLDT_UINT32, &p->class_, sizeof(p->class_), NULL, NULL) ||
```

Comment:
Can you double-check the types of these fields? They're pretty much all signed and have platform-dependent sizes, so most of these should be changed to `SQLDT_INT`, `SQLDT_SHORT`, etc. (matching the actual data type in the struct)

Reply:
Yes, our sql code is a bit messy and not very well documented, unfortunately.

The `SqlStmt_BindColumn()` and `SqlStmt_BindParam()` functions (through `Sql_P_BindSqlDataType()` set the `buffer_type` and `is_unsigned` fields of the `MYSQL_BIND` struct that is later passed to `mysql_stmt_fetch_column()` / `mysql_stmt_bind_param()`.

The [MySQL C API documentation](https://dev.mysql.com/doc/refman/8.0/en/c-api-prepared-statement-type-codes.html) for those `MYSQL_TYPE_` constants used mentions:

> The buffer_type member of MYSQL_BIND structures indicates the data type of the C language variable bound to a statement parameter or result set column. For input, buffer_type indicates the type of the variable containing the value to be sent to the server. For output, it indicates the type of the variable into which a value received from the server should be stored.

(and it also shows which types are allowed for the different SQL column types.

We should probably update our documentation to mention this...

Reply:
Okay, so for input/output data always use the SQL data type corresponding to the C data type. That's described precisely.

But what about params used for comparison, like `pet_id` in this statement?
That's not mentioned in the linked API documentation. (Or I just didn't notice it.)

Reply:
Done.